infra/presubmit.py

```diff
@@ -345,32 +340,49 @@ def get_changed_files():
   return [os.path.abspath(f) for f in changed_files]
 
 
-def is_test_dir_blocklisted(directory):
-  """Returns True if |directory| is blocklisted."""
-  for blocklisted_regex in TEST_BLOCKLIST:
-    if blocklisted_regex.search(directory):
-      return True
-  return False
+def run_build_tests():
+  """Runs build tests because they can't be run in parallel."""
+  suite_list = [
+      unittest.TestLoader().discover(os.path.join(_SRC_ROOT, 'infra', 'build'),
+                                     pattern='*_test.py'),
+  ]
+  suite = unittest.TestSuite(suite_list)
+  result = unittest.TextTestRunner().run(suite)
+  success = not result.failures and not result.errors
+  if not success:
+    print('Build tests failed.')
+  return success
 
 
-def run_tests(_=None):
-  """Run all unit tests."""
+def run_nonbuild_tests(parallel):
+  """Run all tests but build tests. Do it in parallel if |parallel|. The reason
+  why we exclude build tests is because they use an emulator that prevents them
+  from being used in parallel."""
+  # We look for all project directories because otherwise pytest won't run tests
+  # that are not in valid modules (e.g. "base-images").
   relevant_dirs = set()
   all_files = get_all_files()
   for file_path in all_files:
     directory = os.path.dirname(file_path)
-    if is_test_dir_blocklisted(directory):
-      continue
     relevant_dirs.add(directory)
 
-  suite_list = []
-  for relevant_dir in relevant_dirs:
-    suite_list.append(unittest.TestLoader().discover(relevant_dir,
-                                                     pattern='*_test.py'))
-  suite = unittest.TestSuite(suite_list)
-  result = unittest.TextTestRunner().run(suite)
+  # Use ignore-glob because ignore doesn't seem to work properly with the way we
+  # pass directories to pytest.
+  command = [
+      'pytest',
+      '--ignore-glob=infra/base-images/base-sanitizer-libs-builder/*',
+      '--ignore-glob=infra/build/*',
+  ]
+  if parallel:
+    command.extend(['-n', 'auto'])
+  command += list(relevant_dirs)
+  return subprocess.run(command, check=False).returncode == 0
```

Comment:
Should we show why these failed, like print output or something.
Also, should we print something at start of function to tell "Running build tests", "Running non-build tests", etc.

Reply:
>Should we show why these failed, like print output or something.

That happens already.

>Also, should we print something at start of function to tell "Running build tests", "Running non-build tests", etc.

Done